packages/grafanaPlugin/default.nix

```diff
@@ -0,0 +1,37 @@
+{ pkgs, plugins, fetchurl ? pkgs.fetchurl, unzip ? pkgs.unzip }:
+
+let
+  fetchPlugins = { name, version ? "latest", ... } @ args: (fetchurl ({
+    inherit name;
+    url = "https://grafana.com/api/plugins/${name}/versions/${version}/download";
+    recursiveHash = true;
+    downloadToTemp = true;
+    postFetch = ''
+      unpackDir="$TMPDIR/unpack"
+      mkdir "$unpackDir"
+      cd "$unpackDir"
+      renamed="$TMPDIR/${name}.zip"
+      mv "$downloadedFile" "$renamed"
+      unpackFile "$renamed"
```

Comment:
Where is unpackFile defined? It might make more sense to just write your own command here. It will remove the need for overrideAttrs. Something like:
```
mkdir -p $out
${pkgs.unzip}/bin/unzip -d $out $renamed
```

Reply:
unpack file is defined in setup https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L836 
im unsure whats the more standard way I based this of fetch zip

Reply:
tested your solution and it works well